cmd/ctl/check_config.go

```diff
@@ -0,0 +1,376 @@
+	parsed, err := x509FromTLSSecret(ctx, kubeClient, ns, name)
+	if err != nil {
+		return err
+	}
+	if time.Now().After(parsed.NotAfter) {
+		return fmt.Errorf("certificate in secret %s/%s is expired", ns, name)
+	}
+	return nil
+}
+
+func jwtKeyValid(ctx context.Context, kubeClient kubernetes.Interface, ns, name string) error {
+	key, err := tlsutil.JWTSigningKeyFromSecret(ctx, kubeClient, ns, name)
+	if err != nil {
+		return err
+	}
+	// Require RSA for now
+	if _, ok := key.(*rsa.PrivateKey); !ok {
+		return fmt.Errorf("JWT signing key is not an RSA private key")
+	}
+	return nil
+}
+
+// verifyArgoCDClusterScoped checks operator CR or argocd-cm to ensure no namespaced mode is configured.
+func verifyArgoCDClusterScoped(ctx context.Context, kc *kube.KubernetesClient, ns string) error {
+	// Try operator CR first: group argoproj.io, version v1alpha1, resource argocds
+	gvr := schema.GroupVersionResource{Group: "argoproj.io", Version: "v1alpha1", Resource: "argocds"}
+	// If CRD exists and an ArgoCD instance exists in ns, assert spec.applicationNamespaces is empty
+	if kc.DynamicClient != nil {
+		list, err := kc.DynamicClient.Resource(gvr).Namespace(ns).List(ctx, metav1.ListOptions{})
+		if err == nil && list != nil && len(list.Items) > 0 {
+			for _, it := range list.Items {
+				if arr, found, _ := unstructured.NestedSlice(it.Object, "spec", "applicationNamespaces"); found && len(arr) > 0 {
+					return fmt.Errorf("argo CD configured for namespaced mode via spec.applicationNamespaces")
+				}
+			}
+			return nil
+		}
+	}
+	// Fallback when Operator CR is absent: check the core Argo CD ConfigMap (argocd-cm)
+	// and fail if application.namespaces is set (namespaced mode). If the ConfigMap
+	// does not exist or the key is empty, assume cluster-scoped and pass.
+	cm, err := kc.Clientset.CoreV1().ConfigMaps(ns).Get(ctx, "argocd-cm", metav1.GetOptions{})
+	if err == nil {
+		if v := strings.TrimSpace(cm.Data["application.namespaces"]); v != "" {
+			return fmt.Errorf("argocd-cm sets application.namespaces (%s)", v)
+		}
+		return nil
+	}
+	// If neither CR nor ConfigMap found, do not fail the check
+	return nil
+}
+
+// verifyRouteHostMatchesCert checks OpenShift Route host in ns is present in IPS?DNS of the given TLS secret.
+func verifyRouteHostMatchesCert(ctx context.Context, kc *kube.KubernetesClient, ns string, tlsSecretName string) error {
+	// Discover route API by attempting a list via dynamic client
+	if kc.DynamicClient == nil {
+		return nil // skip
+	}
+	// Load cert IPS/DNS
+	parsed, err := x509FromTLSSecret(ctx, kc.Clientset, ns, tlsSecretName)
+	if err != nil {
+		return err
+	}
+
+	// Try routes.route.openshift.io
+	gvr := schema.GroupVersionResource{Group: "route.openshift.io", Version: "v1", Resource: "routes"}
+	routes, err := kc.DynamicClient.Resource(gvr).Namespace(ns).List(ctx, metav1.ListOptions{})
+	if err != nil || routes == nil {
+		return nil // skip if Route API not present or inaccessible
```

Comment:
If the Route API doesn't exist, we can return nil, but for any other error we should return the error rather than ignoring it.